docs/setup-project.md

```diff
@@ -0,0 +1,176 @@
+# Setting up a ChugSplash project
+
+This guide will show you how to setup a project with the ChugSplash Hardhat plugin.
+
+If your repository has an existing Hardhat project, you can skip to [here](#install-chugsplash).
+
+## Table of Contents
+
+- [Initialize Node.js](#initialize-nodejs)
+- [Setup Hardhat](#setup-hardhat)
+- [Install ChugSplash](#install-chugsplash)
+- Setup ChugSplash:
+  - [In a JavaScript Project](#setup-chugsplash-using-javascript)
+  - [In a TypeScript Project](#setup-chugsplash-using-typescript)
+- [Learn More](#learn-more)
+
+## Initialize Node.js
+
+If you are in a new repository, initialize a Node.js project using [yarn](https://classic.yarnpkg.com/lang/en/) or [npm](https://docs.npmjs.com/cli/v8/). Otherwise, skip to the next step.
+
+With Yarn:
+```
+yarn init -y
+```
+
+With npm:
+```
+npm init -y
+```
+
+## Setup Hardhat
+
+Install Hardhat in your repository if it isn't already installed.
+
+With Yarn:
+```
+yarn add --dev hardhat
+```
+
+With npm:
+```
+npm install --save-dev hardhat
+```
+
+Then, initialize Hardhat:
+
+```
+npx hardhat
+```
+
+## Install ChugSplash
+
+With Yarn:
+```
+yarn add --dev @chugsplash/plugins @chugsplash/core
+```
+With npm:
+```
+npm install --save-dev @chugsplash/plugins @chugsplash/core
+```
+
+## Setup ChugSplash using JavaScript
```

Comment:
I'm conflicted about duplicating this info for both Javascript and Typescript. I feel like we could get away with just including typescript and then note that the `chugsplash-init` command will also generate javascript depending on your project setup.

Reply:
Yeah it's not super elegant but I think it's the best solution for now. We could probably get away with just having a single TS or JS guide, but I think this provides a subpar experience for users who are trying to setup with the language that we haven't explicitly given instructions for. Fwiw Hardhat has a [JS guide](https://hardhat.org/hardhat-runner/docs/guides/project-setup) and a [TS guide](https://hardhat.org/hardhat-runner/docs/guides/typescript), and so does [smock](https://smock.readthedocs.io/en/latest/getting-started.html)